src/dscigametrics/compute_metrics.py

```diff
@@ -30,4 +33,62 @@ def compute_metrics(data, campaign_id, start_date, end_date):
         Conversion rate: 0.23
         Total transaction revenue: $100
         Average transaction revenue: $2.33"
-    """
\ No newline at end of file
+    """
+
+    if not isinstance(data, pd.DataFrame):
+        raise TypeError('Your data argument must be a pandas dataframe')
+    if not isinstance(campaign_id, int):
+        raise TypeError('The campaign ID should be an integer')
+    if not isinstance(start_date, int):
+        raise TypeError('Dates are entered as integers. 1st August, 2022 should be 20220801')
+    if not isinstance(end_date, int):
+        raise TypeError('Dates are entered as integers. 1st August, 2022 should be 20220801')
+    
+    df_filter_campaign = data.loc[(data['trafficSource.adwordsClickInfo.campaignId']==campaign_id)]
+    df_filter_start = df_filter_campaign.loc[(df_filter_campaign['date'] >= start_date)]
+    df_filtered = df_filter_start.loc[(df_filter_start['date'] <= end_date)]
+
+    if df_filtered.shape[0] < 1:
+        raise ValueError('There is no data matching your query')
+    
+    def total_transaction_revenue(df):
+
+        revenue = df['totals.transactionRevenue'].sum()
+
+        return revenue
+
+    def average_transaction_revenue(df):
+        
+        transactions_count = df['totals.transactions'].sum()
+        revenue = df['totals.transactionRevenue'].sum()
+
+        return revenue/transactions_count
```

Comment:
There might be a `ZeroDivisionError` when there's no transaction at all. That is, when`df['totals.transactions'].sum() == 0`.
I might choose either of these 2 methods:
- Using the function `mean()` to calculate the average and avoid the error when there's no transaction. -> `df['totals.transactionRevenue'].mean()`
- Using `if-else` to determine whether `transactions_count == 0` beforehand. If it is 0, then return 0; else, move forward to `revenue / transactions_count`.

Reply:
That's a really good point! I've updated it